tests/Infection/TrinaryLogicMutator.php

```diff
@@ -0,0 +1,66 @@
+<?php declare(strict_types=1);
+
+namespace PHPStan\Infection;
+
+use Infection\Mutator\Definition;
+use Infection\Mutator\Mutator;
+use Infection\Mutator\MutatorCategory;
+use LogicException;
+use PhpParser\Node;
+
+/**
+ * @implements Mutator<Node\Expr\MethodCall>
+ */
+final class TrinaryLogicMutator implements Mutator {
+
+	public static function getDefinition(): Definition
+	{
+		return new Definition(
+			<<<'TXT'
+                Replaces TrinaryLogic->yes() with !TrinaryLogic->no()
+                TXT
+			,
+			MutatorCategory::ORTHOGONAL_REPLACEMENT,
+			null,
+			<<<'DIFF'
+                - $type->isBoolean()->yes();
+                + !$type->isBoolean()->no();
+                DIFF,
+		);
+	}
+
+	public function getName(): string
+	{
+		return 'TrinaryLogicMutator';
+	}
+
+	public function canMutate(Node $node): bool
```

Comment:
atm this class replaces any call to a method named `yes`, or `no`.
for now it is not scoped to the `TrinaryLogic` class.

Reply:
atm this mutator works for these classes in the phpstan-src codebase - which I think is not a bad thing 

<img width="652" height="259" alt="grafik" src="https://github.com/user-attachments/assets/6b69235f-c397-4fa1-9354-a3e07bac6bd9" />